src/main/java/org/opensearch/knn/plugin/transport/TrainingModelRequest.java

```diff
@@ -103,6 +107,11 @@ public TrainingModelRequest(StreamInput in) throws IOException {
         this.maximumVectorCount = in.readInt();
         this.searchSize = in.readInt();
         this.trainingDataSizeInKB = in.readInt();
+        if (IndexUtil.isVersionOnOrAfterMinRequiredVersion(in.getVersion(), KNNConstants.MODEL_VECTOR_DATA_TYPE_KEY)) {
+            this.vectorDataType = VectorDataType.get(in.readOptionalString());
```

Comment:
Remove Optional here - for clusters above the min version, they should have it. Also, do same for writing.

Reply:
Ack